packages/nx/src/native/tasks/hash_planner.rs

```diff
@@ -0,0 +1,352 @@
+use crate::native::tasks::{dep_outputs::get_dep_output, types::TaskGraph};
+use crate::native::types::{Input, NxJson};
+use crate::native::{
+    project_graph::types::ProjectGraph,
+    tasks::{inputs::SplitInputs, types::Task},
+};
+use rayon::prelude::*;
+use std::collections::HashMap;
+
+use crate::native::tasks::inputs::{
+    expand_single_project_inputs, get_inputs, get_inputs_for_dependency, get_named_inputs,
+};
+use crate::native::tasks::utils;
+use crate::native::utils::find_matching_projects;
+
+#[napi]
+pub struct HashPlanner {
+    nx_json: NxJson,
+    project_graph: ProjectGraph,
+    workspace_root: String,
+}
+
+#[napi]
+impl HashPlanner {
+    #[napi(constructor)]
+    pub fn new(workspace_root: String, nx_json: NxJson, project_graph: ProjectGraph) -> Self {
+        Self {
+            nx_json,
+            project_graph,
+            workspace_root,
+        }
+    }
+
+    #[napi]
+    pub fn get_plans(
+        &self,
+        task_ids: Vec<&str>,
+        task_graph: TaskGraph,
+    ) -> anyhow::Result<HashMap<String, Vec<String>>> {
```

Comment:
Create a separate function which keeps it as a `HashInstruction` which will be easier to work with within Rust.

Reply:
enum HashInstruction {
  'AllExternalDependencies'
  'ExternalNode'
  'FileSet'
  'EnvironmentVariable'
  'Runtime'
  'ProjectConfiguration'
  'TsConfiguration'
  'TaskOutput'
}